### PR TITLE
Qt1-442 bulk upload log edit

### PR DESCRIPTION
**Pull request by alireid**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddAndApproveDraftApprenticeshipsRequestMapper.cs [hunk 1]
```diff
@@ -22,10 +22,13 @@ public async Task<BulkUploadAddAndApproveDraftApprenticeshipsRequest> Map(FileUp
         {
             var cacheModel = await _cacheService.GetFromCache<FileUploadCacheModel>(source.CacheRequestId.ToString());
             var csvRecords = cacheModel.CsvRecords;
+            var fileUploadLogId = cacheModel.FileUploadLogId;
+
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddAndApproveDraftApprenticeshipsRequestMapper.cs:26]: No test for this, but considering there are no tests for this class, I think we are allowed to ignore this

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddDraftApprenticeshipsRequestMapper.cs [hunk 2]
```diff
@@ -26,10 +27,13 @@ public async Task<BulkUploadAddDraftApprenticeshipsRequest> Map(FileUploadReview
         {
             var cacheModel = await _cacheService.GetFromCache<FileUploadCacheModel>(source.CacheRequestId.ToString());
             var csVRecords = cacheModel.CsvRecords;
+            var fileUploadLogId = cacheModel.FileUploadLogId;
+
             await _cacheService.ClearCache(source.CacheRequestId.ToString(),nameof(FileUploadReviewViewModelToBulkUploadAddDraftApprenticeshipsRequestMapper));
             return new BulkUploadAddDraftApprenticeshipsRequest
             {
                 ProviderId = source.ProviderId,
+                FileUploadLogId = fileUploadLogId,
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddDraftApprenticeshipsRequestMapper.cs:36]: WhenMappingFileUploadStartViewModelToBulkUploadRequestMapperTests.cs

is the testing class for this mapper. We should add a test in to ensure the log id is mapped as expected

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/IOuterApiService.cs [hunk 3]
```diff
@@ -33,6 +33,7 @@ public interface IOuterApiService
         Task<GetPriorLearningSummaryQueryResult> GetPriorLearningSummary(long providerId, long cohortId, long apprenticeshipId);
         Task<GetCohortDetailsResponse> GetCohortDetails(long providerId, long cohortId);
         Task<long> CreateFileUploadLog(long providerId, IFormFile attachment, List<CsvRecord> csvRecords);
+        Task<long> UpdateFileUploadLog(long providerId, long logId);
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/IOuterApiService.cs:36]: I don't think we are using this code yet, and writing code before it's needed causes confusion in this already messy codebase. We may never need this endpoint depending on how we implement that unhandled error exception. Also the parameters here are just identity params and there is no update content. Would it be easier just to remove it

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/OuterApiService.cs [hunk 4]
```diff
@@ -128,7 +128,19 @@ public async Task<long> CreateFileUploadLog(long providerId, IFormFile attachmen
...
+        public async Task<long> UpdateFileUploadLog(long providerId, long logId)
+        {
+            var request = new FileUploadUpdateLogRequest
+            {
+                ProviderId = providerId,
+                LogId = logId,
+            };
+
+            var response = await _outerApiClient.Post<FileUploadUpdateLogResponse>(new PostFileUploadUpdateLogRequest(request));
+            return response.LogId;
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/OuterApiService.cs:143]: As above

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Requests/FileUploadUpdateLogRequest.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+
+namespace SFA.DAS.ProviderCommitments.Infrastructure.OuterApi.Requests
+{
+    public class FileUploadUpdateLogRequest : ApimSaveDataRequest
+    {
+        public long ProviderId { get; set; }
+        public long LogId { get; set; }
+    }
+}
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Requests/FileUploadUpdateLogRequest.cs:10]: we could remove this (as it's not yet needed)

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Requests/PostFileUploadUpdateLogRequest.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+        public string PostUrl => "BulkUpload/UpdateLog";
+
+        public object Data { get; set; }
+
+        public PostFileUploadUpdateLogRequest(FileUploadUpdateLogRequest data)
+        {
+            Data = data;
+        }
+    }
+}
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Requests/PostFileUploadUpdateLogRequest.cs:14]: And this

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Responses/FileUploadLogResponse.cs [hunk 7]
```diff
@@ -2,6 +2,6 @@
 {
     public class FileUploadLogResponse
     {
-        public long FileUploadLogId { get; set; }
+        public long LogId { get; set; }
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Responses/FileUploadLogResponse.cs:5]: and this, but would we ever what a response back? We already know teh Id

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Responses/FileUploadUpdateLogResponse.cs [hunk 8]
```diff
@@ -0,0 +1,7 @@
+﻿namespace SFA.DAS.ProviderCommitments.Infrastructure.OuterApi.Responses
+{
+    public class FileUploadUpdateLogResponse
+    {
+        public long LogId { get; set; }
+    }
+}
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Responses/FileUploadUpdateLogResponse.cs:7]: As above

**Review by Pauljgraham — COMMENTED** [review]
Looks very good and works, just a few tiny comments (and one slightly bigger one about removing the unused code)

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/IOuterApiService.cs (hunk 3 above)
Comment (alireid) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/IOuterApiService.cs:36]: hi paul, removed all the update endpoint stuff now

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Responses/FileUploadLogResponse.cs (hunk 7 above)
Comment (alireid) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Responses/FileUploadLogResponse.cs:5]: hi paul, agreed, all update stuff removed

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Responses/FileUploadUpdateLogResponse.cs (hunk 8 above)
Comment (alireid) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Responses/FileUploadUpdateLogResponse.cs:7]: hi paul, all update stuff removed now

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Requests/PostFileUploadUpdateLogRequest.cs (hunk 6 above)
Comment (alireid) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Requests/PostFileUploadUpdateLogRequest.cs:14]: removed as dumped all update endpoint stuff

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Requests/FileUploadUpdateLogRequest.cs (hunk 5 above)
Comment (alireid) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/Requests/FileUploadUpdateLogRequest.cs:10]: yep removed as part of mass removal of update endpoint

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/OuterApiService.cs (hunk 4 above)
Comment (alireid) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Infrastructure/OuterApi/OuterApiService.cs:143]: removed as part of update endpoint delete

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddDraftApprenticeshipsRequestMapper.cs (hunk 2 above)
Comment (alireid) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddDraftApprenticeshipsRequestMapper.cs:36]: Added a test for this based on asset from cache id to apirequest param

**Review by Pauljgraham — APPROVED** [review]

**Pull request merged**